crates/server/src/server/server_future.rs

```diff
@@ -733,6 +733,14 @@ impl<T: RequestHandler> ServerFuture<T> {
         block_until_done(&mut self.join_set).await
     }
 
+    /// Returns a clone of the [`CancellationToken`] used to gracefully shut down the server.
+    ///
+    /// Once cancellation is requested, all background tasks will stop accepting new connections,
+    /// and `block_until_done()` will complete once all tasks have terminated.
+    pub fn shutdown_token(&self) -> CancellationToken {
```

Comment:
I think it is cleaner to yield `&CancellationToken`, leaving the decision to clone to the caller (especially given that `CancellationToken` has a bunch of `&self` methods, so just the reference might be useful).